synapse/models/files.py

```diff
@@ -183,6 +183,40 @@ def getModelDefs(self):
                     'ex': 'c:/windows/system32/calc.exe'}),
             ),
 
+            'interfaces': (
+                ('file:mime:meta', {
+                    'props': (
+                        ('file', ('file:bytes', {}), {
+                            'doc': 'The file that the mime info was parsed from.'}),
+                        ('file:offs', ('int', {}), {
+                            'doc': 'The optional offset where the mime info was parsed from.'}),
+                        ('file:data', ('data', {}), {
+                            'doc': 'A mime specific arbitrary data structure for non-indexed data.',
+                        }),
+                    ),
+                    'doc': 'Properties common to mime specific file metadata types.',
+                }),
+                ('file:mime:msoffice', {
+                    'props': (
+                        ('title', ('str', {}), {
+                            'doc': 'The title extracted from Microsoft Office metadata.'}),
+                        ('author', ('str', {}), {
+                            'doc': 'The author extracted from Microsoft Office metadata.'}),
+                        ('subject', ('str', {}), {
+                            'doc': 'The subject extracted from Microsoft Office metadata.'}),
+                        ('application', ('str', {}), {
+                            'doc': 'The creating_application extracted from Microsoft Office metadata.'}),
+                        ('created', ('time', {}), {
+                            'doc': 'The create_time extracted from Microsoft Office metadata.'}),
+                        ('lastsaved', ('time', {}), {
+                            'doc': 'The last_saved_time extracted from Microsoft Office metadata.'}),
+                    ),
+                    'doc': 'Properties common to various microsoft office file formats.',
+                    # yo dawg...
+                    'interfaces': ('file:mime:meta',),
```

Comment:
Are we going to have documentation reqs to note ifaces separately ?

Reply:
Eventually, yes, but the interfaces cause the model to include the properties defined for each model so until we (incrementally) implement interface awareness in the docs ( and other things ) it will behave correctly.